File: src/test/programmers/java/level2/FunctionDevelopment.java
package test.programmers.java.level2;

import java.util.*;

public class FunctionDevelopment {
    public static void main(String[] args) {
        int[] p = new int[]{93, 30, 55};
        int[] s = new int[]{1, 30, 5};

        System.out.println(Arrays.toString(solution(p, s)));
    }
    static public int[] solution(int[] progresses, int[] speeds) {
        LinkedList<Integer> queue = new LinkedList<>();
        List<Integer> temp_answer = new ArrayList<>();

        for(int i = 0; i < progresses.length; i++) {
            int time = (int)(Math.ceil((100 - progresses[i]) / (float)speeds[i]));

            while(!queue.isEmpty() && queue.peekFirst() < time) {
                temp_answer.add(queue.size());
                queue.clear();
            }

            queue.offer(time);
        }

        temp_answer.add(queue.size());

        int[] answer = temp_answer.stream().mapToInt(i -> i).toArray();
        return answer;
    }
}

File: src/test/baekjoon/python/binarysearch/FunctionDevelopment.py
def solution(progresses, speeds):
    answer = []
    time, count = 0, 0
    while progresses:
        if progresses[0] + time * speeds[0] >= 100:
            count += 1
            progresses.pop(0)
            speeds.pop(0)
        else:
            if count != 0:
                answer.append(count)
                count = 0
            time += 1
    answer.append(count)
    return answer

print(solution([93, 30, 55], [1, 30, 5]))
